newhelm/single_turn_prompt_response.py

```diff
@@ -1,15 +1,28 @@
-from typing import Any, Dict, List, Type, TypeVar
+from typing import Any, Dict, List, Mapping, Optional, Type, TypeVar, Union
 
 from pydantic import BaseModel, Field
 from newhelm.annotation import Annotation
 
 from newhelm.placeholders import Prompt
 from newhelm.sut import SUTResponse
+from newhelm.typed_data import TypedData
 
 # TODO: This whole file assumes single turn. We'll either need to make it
 # more complicated, or make parallel structures for multi-turn.
 
-_InT = TypeVar("_InT")
+_BaseModelType = TypeVar("_BaseModelType", bound=BaseModel)
+_Context = Optional[Union[TypedData, str, Mapping]]
+
+
+def resolve_context_type(context: _Context, cls):
```

Comment:
Was this meant to be `_resolve_context_type`?

Reply:
I thought it was plausible that someone would want to call this on their own. I don't have a strong opinion though, so I'm happy making it private if you want.

Reply:
Totally cool with it being public. Was just a comment to make sure it wasn't accidental.